src/neo/SmartContract/Native/Oracle/OracleContract.cs

```diff
@@ -145,19 +145,23 @@ protected override void PostPersist(ApplicationEngine engine)
                 OracleResponse response = tx.GetAttribute<OracleResponse>();
                 if (response is null) continue;
 
+                // Check tx state
+                if (engine.Snapshot.Transactions.TryGet(tx.Hash)?.VMState != VMState.HALT) continue;
```

Comment:
But it can legitimately fail. For example, oracles errored out the request and bring an error status to the contract. The contract is not happy about that (it's expecting a successful response) and `ABORT`s the execution. Transaction fails, but oracles did their job of bringing response to the chain, they should receive their GAS for it.

Reply:
Why they should be rewarded with `FAULT` tx?

Reply:
_“Transaction fails, but oracles did their job of bringing response to the chain, they should receive their GAS for it.”_

It‘s impossible to distinguish between the failure of the transaction execution, due to the evil of the oracle nodes and its own contract problems, such as insufficient fee or different execution logic.

Reply:
Because they did their job. The callback contract invoked from `finish` is free to do whatever it wants to with the data provided by oracles, including failing the execution, but oracles did provide that data.

Reply:
Done, it have sense, but I will think more about this